src/Storages/StorageMergeTree.cpp

```diff
@@ -1969,8 +1972,6 @@ void StorageMergeTree::dropPartition(const ASTPtr & partition, bool detach, Cont
         }
     }
 
-    /// Old parts are needed to be destroyed before clearing them from filesystem.
```

Comment:
Shouldn't the same be done for `StorageMergeTree::dropPart` too? If not, then I don't see why.

Reply:
Sure, missed that, thanks. Rebased